devenv.nix

```diff
@@ -54,6 +54,12 @@
     entry = "bun run db:generate-types";
   };
   
+  pre-commit.hooks.typecheck = {
```

Comment:
see [this comment](https://github.com/MakePrisms/boardwalkcash/pull/331/files#r2049068109) for why I added this